mage_ai/orchestration/pipeline_scheduler.py

```diff
@@ -253,6 +254,7 @@ def schedule(self, block_runs: List[BlockRun] = None) -> None:
             elif self.pipeline.run_pipeline_in_one_process:
                 self.__schedule_pipeline()
             else:
+                self.__check_block_run_timeout()
```

Comment:
If certain block runs time out and get updated to failed, we should not proceed with `__schedule_blocks`.
You can just return here and let it check `(self.pipeline_run.any_blocks_failed() and not self.allow_blocks_to_fail)` next time the `schedule` method is called.